bin/sl-deploy.js

```diff
@@ -93,10 +107,27 @@ if (!local)
       return exit(err);
     }
     debug('Connecting to %s via %s', baseURL, url);
-    deploy(process.cwd(), url, branchOrPack, config, exit);
+    var client = new MeshClient(url);
+    client.serviceFindOrCreate(serviceName, 1, function(err, service) {
+      if (err) {
+        if (err.statusCode === 401) {
+          console.error(
+            'Cannot access remote. If authentication is required,' +
+            ' credentials should be given in the URL.');
+        } else {
+          console.error('Error connecting to server:', err);
+        }
+        return exit(err);
+      }
+      deploy(
+        process.cwd(), service.getDeployEndpoint(),
```

Comment:
could be `workingDir`, current code is fine by me, but I noticed you replaced another process.cwd() with workingDir.